include/CL/sycl/math.hpp

```diff
@@ -167,6 +170,129 @@ TRISYCL_MATH_WRAP(trunc)
 //*TRISYCL_MATH_WRAP3(mad24)
 //*TRISYCL_MATH_WRAP3(mul24)
 
+// Note: A lot of these math functions could be forwarded onto the OpenCL
+// implementations when compiling for OpenCL
+
+// Returns the cross product of p0.xyz and p1.xyz. The w component of float4
+// result returned will be 0.0.
+template <typename T, int size>
+vec<T, size> cross(vec<T, size> p0,
+                   vec<T, size> p1) {
+  static_assert(!(size > 4 || size < 3), "vec type should be 3 or 4 dimensions");
```

Comment:
Remove the negation. Probably `size == 3 || size == 4` is just obviously understandable... :-)

Break the line after `,`